### PR TITLE
fix(aws-crypto): export explicit dependencies on @aws-sdk/types

### PR DESCRIPTION
**Pull request by alexforsyth**
_Issue #, if available:_
https://github.com/aws/aws-sdk-js-v3/issues/1842

_Description of changes:_
Explicitly declare a dependency on `@aws-sdk/types` for all packages that depend on `@aws-sdk` types so that downstream consumers of these packages don't have to install `@aws-sdk/types` themselves. The issue is detailed https://github.com/aws/aws-sdk-js-v3/issues/1842#issuecomment-758184096. 

This impacts:
```
crc32
ie11-detection
random-source-*
sha256-*
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/random-source-node/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (trivikr) [inline packages/random-source-node/package-lock.json:1]: `package-lock.json` shouldn't exist on individual packages.
This project uses package-lock.json at root https://github.com/aws/aws-sdk-js-crypto-helpers/blob/master/package-lock.json

File: packages/random-source-node/package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
   },
   "license": "Apache-2.0",
   "dependencies": {
+    "@aws-sdk/types": "^3.1.0",
```
Comment (trivikr) [inline packages/random-source-node/package.json:21]: nit: standardize using `^3.2.0` as the new version is now released for JS SDK v3.
```suggestion
    "@aws-sdk/types": "^3.2.0",
```

**Comment by seebees** [comment]
The intention here is to roll back #188 right?

**Comment by alexforsyth** [comment]
@seebees I dont believe rolling back #188 is needed #

> The intention here is to roll back #188 right?

File: packages/random-source-node/package-lock.json (hunk 1 above)
Comment (alexforsyth) [reply packages/random-source-node/package-lock.json:1]: Whoops. Nice callout

File: packages/random-source-node/package.json (hunk 2 above)
Comment (alexforsyth) [reply packages/random-source-node/package.json:21]: Yeah I think the `^` will take care of it, but yeah why not just update it to the most recent anyways

Comment (alexforsyth) [reply packages/random-source-node/package.json:21]: I think the most recent version of types is 3.1.0: https://www.npmjs.com/package/@aws-sdk/types
Screen Shot 2021-01-12 at 11 54 53 AM

**Comment by seebees** [comment]
@alexforsyth I was just making sure I understood the intent here. I went digging and found https://github.com/aws/aws-sdk-js-v3/issues/1842 But I wanted to make sure, for history, that there is a clear link between #188 and this.

File: packages/random-source-node/package.json (hunk 2 above)
Comment (trivikr) [reply packages/random-source-node/package.json:21]: Yup, the `v3.2.0` was not released as there were no updates in `@aws-sdk/types`

Verified by running the command:
```
npm view @aws-sdk/types versions
```

**Comment by alexforsyth** [comment]
@seebees yeah #188 was an initial attempt to fix this issue, but after further investigation and discussion among the team this is the correct way to fix things. If needed please message me internally and we can follow up on this there.

**Review by seebees — APPROVED** [review]

**Review by trivikr — APPROVED** [review]

**Comment by trivikr** [comment]
Note: The typo in PR title needs to fixed from `@aws-types` to `@aws-sdk/types`

**Pull request merged**